File: Navigation/ICaptureSessionPageViewModel.vb
Imports WpfApp1.ViewModels

Namespace Navigation
    Public Interface ICaptureSessionPageViewModel
        Inherits IDisposable

        Event EnterLockRequested As EventHandler
        Event ExitLockRequested As EventHandler(Of CaptureLockRequestEventArgs)

        ReadOnly Property IsLocked As Boolean

        Sub OnLockEntered()
        Sub RequestPauseFromView()
        Sub OnViewUnlockCompleted(reason As CaptureUnlockReason)
    End Interface
End Namespace

File: src/ClickSyncMouseTester/Navigation/ICaptureSessionPageViewModel.cs
using ClickSyncMouseTester.ViewModels;
using System;

namespace ClickSyncMouseTester.Navigation;

public interface ICaptureSessionPageViewModel : IDisposable
{
    bool IsLocked { get; }

    event EventHandler EnterLockRequested;

    event EventHandler<CaptureLockRequestEventArgs> ExitLockRequested;

    void OnLockEntered();

    void RequestPauseFromView();

    void OnViewUnlockCompleted(CaptureUnlockReason reason);
}





